DVL model updates and demo

dtdavi1 opened the pull request:
Added all of the DVL models from the "dave" repository into this one and updated everything so that it runs--same launch commands as in the tutorial, but from the nps_uw_sensors_gazebo package (I'll update the wiki tutorial in the next couple of days).

Added a "side-by-side" demo of the 3 meshes that we've got so far.  The demo can be launched with the command "roslaunch nps_uw_sensors_gazebo whoi_dvl_static_side_by_side.launch" and will launch both a Gazebo window and an RVIZ window (the RVIZ window has better lighting).  The plugins are "live" for all 3 models, so the DVL topics are being published.  The models are all attached to the same base link (easier to get into RVIZ), so moving one will probably move all 3.

dtdavi1 committed:
Moved DVL models & demos from "dave" to here for release 1

Path and naming changes to align with this package's organization.

dtdavi1 committed:
Updated DVL demo "robot" to use eNovus mesh and dimensions

Also updated WHOI DVL urdf files to include an "enable_water_tracking"
element in the plugin parameters.  When enabled (1), the DVL plugin
will include water tracking results (based on robot base link motion,
so it won't account for current yet) if the NPS additions to the WHOI
DSL environment are used (it will work fine with the WHOI version).

dtdavi1 committed:
Added a side-by-side visual with the 3 DVL meshes we've got so far

The model includes the Teledyne WHN, Teledyne Explorer and Nortek
DVL1000 and is launched from the command line with
"roslaunch nps_uw_sensors_gazebo whoi_dvl_static_side_by_side.launch"

Gravity is disabled so that the plugins (WHOI version) will publish
data (even though none of the DVLs is moving) without the DVLs
falling to the ground.  An RVIZ visualization is also launched
that actually provides a better lit view of the DVL meshes.

bsb808 commented:
Cool - does that mean we should close [Dave PR #19](https://github.com/Field-Robotics-Lab/dave/pull/19)?

dtdavi1 commented:
> Cool - does that mean we should close [Dave PR #19](https://github.com/Field-Robotics-Lab/dave/pull/19)?

I'd leave it for now, but don't accept it.  I don't see any reason to maintain the Dave versions any longer (too easy to get mismatches that cause trouble), so as soon as this is merged I'll delete all of the DVL stuff from that one.  Once the deletes are committed the merge will close out that thread.

bsb808 commented:
I'll take a look now.  I'm working on the highlights and want to include the DVL models.

bsb808 commented on models/uuvsim_teledyne_whn_urdf/model.config:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<model>
+  <name>teledyne_whn</name>
```
All of the WHN models have the same name, which makes it a bit confusing.

dtdavi1 replied:
Fixed in latest commit (I think).

BruceMty replied:
Looks good.

bsb808 reviewed (commented):
I would recommend creating standalone Gazebo models for each of the DVL models.   This allows us to have the mesh and model.sdf and model.config files in a labeled directory under models.

As an example, see the 3d_lidar model here: https://github.com/osrf/vrx/tree/master/vrx_gazebo/models
The associated `model.sdf` file includes the visual, collision and inertial values.

Then you can re-use the mesh in the xacro or urdf that defines any mounting, joints, plugins, etc.  See the example here: https://github.com/osrf/vrx/blob/master/wamv_gazebo/urdf/sensors/wamv_3d_lidar.xacro

BruceMty commented:
It looks like some of the .xacro files in urdf are about the same except for some physics properties, for example uuvsim_nortek_dvl_1000_300.xacro and uuvsim_nortek_dvl_100_4000.xacro.  Would it be reasonable to define macros for each of these configurations that wrap a common macro?  Or maybe being separate helps pick-and-choose?

BruceMty commented on urdf/uuvsim_teledyne_whn_robot.xacro:
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0"?>
```
Some of these .xacro files have their execute bit set.

dtdavi1 replied:
Fixed in latest commit.

BruceMty replied:
Done.

dtdavi1 committed:
Created SDF models for all of the performer DVLs

Refactoring in response to Brian's input IRT PR#6.

Moved meshes into the SDF models, and modified the URDF (Xacro) files
to use those meshes rather than ones from a single "meshes" directory.

Meshes for the Teledyne WHN, Teledyne Explorer 1000, and Nortek
DVL1000-4000m are correct.  The other models (SDF and URDF) use the
closest match of the 3 meshes that we have at present.  They have
been named according to the DVL for which they're being used, so
as the actual meshes are developed they can replace what's currently
in the model-specific meshes directory and everything will work.

dtdavi1 committed:
CHMOD for some of the Xacro files that had "execute" set

dtdavi1 commented:
> I would recommend creating standalone Gazebo models for each of the DVL models. This allows us to have the mesh and model.sdf and model.config files in a labeled directory under models.
> 
> As an example, see the 3d_lidar model here: https://github.com/osrf/vrx/tree/master/vrx_gazebo/models
> The associated `model.sdf` file includes the visual, collision and inertial values.
> 
> Then you can re-use the mesh in the xacro or urdf that defines any mounting, joints, plugins, etc. See the example here: https://github.com/osrf/vrx/blob/master/wamv_gazebo/urdf/sensors/wamv_3d_lidar.xacro

Created SDF models for all of the performer DVLs.  Modified all of the Xacro files to use the SDF model meshes.  See the [commit](https://github.com/Field-Robotics-Lab/nps_uw_sensors_gazebo/pull/6/commits/1352f18d2d8b00760670acee446ce7640eccbbf1) message for a more thorough description.

dtdavi1 commented:
>> It looks like some of the .xacro files in urdf are about the same except for some physics properties, for example uuvsim_nortek_dvl_1000_300.xacro and uuvsim_nortek_dvl_100_4000.xacro. Would it be reasonable to define macros for each of these configurations that wrap a common macro? Or maybe being separate helps pick-and-choose?

>Probably.  There are a lot of things that can probably be refactored or simplified.  It's probably too big a change for release 1, so it will need to go onto the to-do list.

Okay, good.

BruceMty commented:
I'm good with this PR.
